projects/ngx-cookie-service-ssr/src/lib/ssr-cookie.service.ts

```diff
@@ -54,6 +56,199 @@ export class SsrCookieService {
     }
   }
 
+  /**
+   * Converts the provided cookie string to a key-value representation.
+   *
+   * @param cookieString - A concatenated string of cookies
+   * @returns Map - Key-value pairs of the provided cookies
+   *
+   * @author: Blake Ballard (blakeoxx)
+   * @since: 16.2.0
+   * @see {@link https://github.com/stevermeister/ngx-cookie-service/blob/f7625d789dc18ea6aebcf136edb4cc01eeac5de9/projects/ngx-cookie-service-ssr/src/lib/ssr-cookie.service.ts#L100}
+   *  for previous implementation of parsing logic
+   */
+   static cookieStringToMap(cookieString: string): Map<string, string> {
```

Comment:
@blakeoxx whats the reason for rewriting the whole logic for all the methods? Can't this be achieved without this?

Reply:
My intent wasn't to rewrite every method. I extracted reusable logic from some of the other methods and moved it to a single method. This was meant to increase maintainability of the code. I probably could've accomplished the feature without this, but it would've meant making changes to several pieces of duplicated code without this refactor.

Reply:
Can you create another PR or update this one with minimal changes? I lost the sight of actual changes with this one

Reply:
Sorry for the late response. I sure can, thanks

Reply:
Sorry for the late response. I sure can, thanks

Reply:
@pavankjadda See #320